### ISSUE TITLE
MAUI won't build with Preview 17.3

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/maui#8360]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/MAUI-wont-build-with-Preview-173/10061656)._

---
Removed all VS4Mac instances and fresh installed. Maui workload was selected. However when I try to build, I get this error:
error NETSDK1147: To build this project, the following workloads must be installed: maui-android
To install these workloads, run the following command: dotnet workload install maui-android.

When I run that command it tells me that it is already installed.
dotnet build on the CLI works

---
#### svaldetero on 6/3/2022, 09:52 AM: 

(private comment, text removed)
#### Feedback Bot on 6/5/2022, 08:16 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by rachelkang** [comment]
Hi there - thanks for opening this issue. .NET MAUI support for VSMac is in the works! In the meantime, you can try installing .NET MAUI via the CLI per the instructions here: https://github.com/dotnet/maui/blob/main/.github/DEVELOPMENT.md

**Comment by svaldetero** [comment]
@rachelkang I'm not sure if some of the information didn't get transferred from the vsfeedback page, but I included that in the original post that it **IS** already installed. The workloads are shown installed via the VS4M installer and can't be changed. The `dotnet` CLI returns that it is installed and trying to reinstall just skips because, it IS installed.

**Comment by jfversluis** [comment]
Sorry about the misunderstanding @svaldetero! Not really sure what happened here, because it seems you reported it in the right place and it got routed to us. If this indeed is a Visual Studio issue then this is not the right place :)

Anyway, let's see if we can get this sorted for you.

**Comment by svaldetero** [comment]
This is a new machine so I haven't added a `global.json` and one doesn't exist in the project. I also do NOT have .NET 7 installed. I'll look through those items and see if they have anything that works for me.

**Comment by svaldetero** [comment]
`maui-check` returns this:
🔎 .NET SDK - Workloads (6.0.301) Checkup...
  ✔ android (Microsoft.NET.Sdk.Android.Manifest-6.0.300 : 32.0.415) installed.
  ❌ ios (Microsoft.NET.Sdk.iOS.Manifest-6.0.300 : 15.4.414) not installed.
  ❌ maccatalyst (Microsoft.NET.Sdk.MacCatalyst.Manifest-6.0.400 : 15.4.414) not installed.
  ❌ tvos (Microsoft.NET.Sdk.tvOS.Manifest-6.0.400 : 15.4.414) not installed.
  ❌ macos (Microsoft.NET.Sdk.macOS.Manifest-6.0.400 : 12.3.414) not installed.
  ✔ maui (Microsoft.NET.Sdk.Maui.Manifest-6.0.300 : 6.0.400) installed.
 `dotnet workload list` returns this:
Installed Workload Ids      Installation Source
-----------------------------------------------
maui-windows                SDK 6.0.300        
maui-maccatalyst            SDK 6.0.300        
maui-ios                    SDK 6.0.300        
maui-android                SDK 6.0.300        
maui                        SDK 6.0.300   

`dotnet --info`
.NET SDK (reflecting any global.json):
 Version:   6.0.301
 Commit:    43f9b18481

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  12.4
 OS Platform: Darwin
 RID:         osx.12-arm64
 Base Path:   /usr/local/share/dotnet/sdk/6.0.301/

Host (useful for support):
  Version: 6.0.6
  Commit:  7cca709db2

.NET SDKs installed:
  6.0.100 [/usr/local/share/dotnet/sdk]
  6.0.300 [/usr/local/share/dotnet/sdk]
  6.0.301 [/usr/local/share/dotnet/sdk]

**Comment by svaldetero** [comment]
I can still build via CLI or with Rider, but VS4Mac complains about the workloads.

**Comment by svaldetero** [comment]
@jfversluis I think I figured out the issue. VS4Mac was trying to use 6.0.100 to build MAUI even though it isn't supported until 6.0.300. I renamed the `6.0.100` folder to `6.0.100.bak` in `/usr/local/share/dotnet/sdk` just to test, and suddenly VS4Mac could build my app again. I'm not sure what changed that caused it to pick the wrong version but it was nothing intentional on my end. 

I don't think this is a MAUI issue, but a VS4Mac issue of not picked the correct SDK; however @samhouts closed the original bug and there's no way to respond now that its closed. I understand that y'all get a ton of bugs to triage but I feel some people are getting trigger happy with the close button.

**Comment by samhouts** [comment]
Not to worry! We can always reopen the Visual Studio bug if we determine it needs to go back that direction. Identifying what version of .NET is used to build the SDK is typically specified by the manifests and templates that the SDK inserts into Visual Studio, which is why this is routed to SDK first.

**Comment by jfversluis** [comment]
Sorry for not engaging more here @svaldetero that was totally my intention, but stuff got in the way. Seems you didn't need me anyway to do an amazing job! Thanks for that! I think you're right about this being a VS4Mac issue.

If I understand correctly you are unblocked now, right Seth?

@samhouts I think we have determined this is indeed for VSMac then? How do we proceed from here? Or do we have another ticket tracking this?

**Comment by svaldetero** [comment]
Correct, I am unblocked now and it does appear to be a VS4Mac issue. Thanks for following up @samhouts and @jfversluis . My opinion is that this one can be closed and other one re-opened unless you have another item (that I'm aware of) that better tracks this.

**Comment by samhouts** [comment]
Done! You can continue tracking the original Developer Community ticket. Thank you!!
